src/Symfony/Component/HttpKernel/EventListener/AbstractSessionListener.php

```diff
@@ -152,17 +152,15 @@ public function onSessionUsage(): void
             return;
         }
 
-        if (!$requestStack = $this->container && $this->container->has('request_stack') ? $this->container->get('request_stack') : null) {
-            return;
+        if ($dataCollector = $this->container && $this->container->has('data_collector.request') ? $this->container->get('data_collector.request') : null) {
+            $dataCollector->collectSessionUsage();
         }
 
-        $stateless = false;
-        $clonedRequestStack = clone $requestStack;
-        while (null !== ($request = $clonedRequestStack->pop()) && !$stateless) {
-            $stateless = $request->attributes->get('_stateless');
+        if (!$requestStack = $this->container && $this->container->has('request_stack') ? $this->container->get('request_stack') : null) {
```

Comment:
This change is for 5.1, if it is valid.
But I'm not sure it's valid: the previous logic was able to detect when a subrequest is declared as stateless while the session is used inside of it. It's legit to be able to nest a stateless subrequest into a stateful one (the reverse is not of course).

Reply:
Agreed!